### PR TITLE
Add infracost integration

### PR DESCRIPTION
**Pull request by agardnerIT**
Add [infracost](https://infracost.io) integration instructions.

Uses job executor service.

Signed-off-by: agardnerit <adam@agardner.net>

### CODE REVIEW

File: infracost/1.0.0/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+kubectl -n YOUR_NAMESPACE create secret generic infracost-details \
+--from-literal=INFRACOST_API_KEY=YOUR_INFRACOST_API_KEY
+```
+
+## Add Necessary Files
+In your Git upstream, add the following files to your stage branch (eg. `dev`). These should be placed inside the service folder.
+
+For example: `service1/job/config.yaml` on the `dev` branch.
+
+- [Job Executor Config File](https://github.com/agardnerIT/keptninfracost2/blob/dev/service1/job/config.yaml)
```
Comment (TannerGabriel) [inline infracost/1.0.0/README.md:46]: I'm getting a 404 error from the link. Can you please take a look?

File: infracost/1.0.0/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- [Terraform plan.json file](https://github.com/agardnerIT/keptninfracost2/blob/dev/service1/files/plan.json)
```
Comment (TannerGabriel) [inline infracost/1.0.0/README.md:47]: I'm getting a 404 error from the link. Can you please take a look?

File: infracost/1.0.0/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Example Shipyard
+
+[Here is an example Shipyard file.](https://github.com/agardnerIT/keptninfracost2/blob/master/shipyard.yaml)
```
Comment (TannerGabriel) [inline infracost/1.0.0/README.md:51]: I'm getting a 404 error from the link. Can you please take a look?

File: infracost/1.0.0/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+4. Sends an `sh.keptn.event.checkcost.started` event to Keptn
+5. Runs `infracost breakdown --path /keptn/files/plan.json --format table`
+6. Sends an `sh.keptn.event.checkcost.finished` event to Keptn, with the tabular output
+
+## Install Job Executor Service
+Follow the [Job executor service install instructions](https://github.com/keptn-contrib/job-executor-service#quickstart) to install on a cluster (either local to Keptn control plane components or on a different cluster).
+
+Ensure the Job Executor Service is installed listening for the `sh.keptn.event.taskname.triggered` event that corresponds to your Shipyard file (see below).
+
+Below, the task is `checkcost` so JES should listen for `sh.keptn.event.checkcost.triggered`.
```
Comment (TannerGabriel) [inline infracost/1.0.0/README.md:29]: Does this add anything that the line above did not already mention?

File: infracost/1.0.0/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+```
+kubectl -n YOUR_NAMESPACE create secret generic infracost-details \
+--from-literal=INFRACOST_API_KEY=YOUR_INFRACOST_API_KEY
+```
+
+## Add Necessary Files
+In your Git upstream, add the following files to your stage branch (eg. `dev`). These should be placed inside the service folder.
+
+For example: `service1/job/config.yaml` on the `dev` branch.
```
Comment (TannerGabriel) [inline infracost/1.0.0/README.md:44]: Maybe you can add the file directly into a code block (if it is not too long)

**Review by TannerGabriel — CHANGES_REQUESTED** [review]
Hey Adam, fantastic integration. There are some minor mistakes as described in my comments.

File: infracost/1.0.0/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# Infracost Integration with Keptn
+
+![infracost running in keptn](./assets/infracost-keptn.png)
```
Comment (TannerGabriel) [inline infracost/1.0.0/README.md:3]: Relative paths are currently not supported (see [here](https://github.com/keptn-contrib/artifacthub#adding-images-to-the-readme-or-artifacthub-pkgyml-file))

File: infracost/1.0.0/artifacthub-pkg.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Artifact Hub package metadata file
+# https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml
+version: 1.0.0
+name: infracost
+displayName: Infracost Integration
+createdAt: 2022-07-23T00:00:00Z
+description: Keptn integration with Infracost
+logoPath: assets/infracost.png
```
Comment (TannerGabriel) [inline infracost/1.0.0/artifacthub-pkg.yml:8]: Relative paths are currently not supported (see [here](https://github.com/keptn-contrib/artifacthub#adding-images-to-the-readme-or-artifacthub-pkgyml-file))

**Review by TannerGabriel — CHANGES_REQUESTED** [review]
Relative path images don't seem to be supported. Could you please change them to absolute paths?

File: infracost/1.0.0/README.md (hunk 5 above)
Comment (agardnerIT) [reply infracost/1.0.0/README.md:44]: Done

File: infracost/1.0.0/README.md (hunk 4 above)
Comment (agardnerIT) [reply infracost/1.0.0/README.md:29]: IMO people scan instructions and miss things so I'm going to opt to leave it in :)

File: infracost/1.0.0/artifacthub-pkg.yml (hunk 7 above)
Comment (agardnerIT) [reply infracost/1.0.0/artifacthub-pkg.yml]: Fixed. I hate this image path issue. Gets me every time!

File: infracost/1.0.0/README.md (hunk 3 above)
Comment (agardnerIT) [reply infracost/1.0.0/README.md]: Fixed by moving files into code rather than linking

**Review by TannerGabriel — APPROVED** [review]
LGTM

**Comment by alikhajeh1** [comment]
@agardnerIT one tiny suggestion: add `--no-color` to the Infracost CLI commands that are run should remove the funky color characters in [this screenshot](https://artifacthub.io/packages/keptn/keptn-integrations/infracost)

**Pull request merged**